ui/helpers/utils/icon-factory.js

```diff
@@ -18,7 +19,7 @@ function IconFactory(jazzicon) {
 IconFactory.prototype.iconForAddress = function (address, diameter) {
   let addr = address;
 
-  if (isHex(address)) {
+  if (isValidHexAddress(address)) {
```

Comment:
rather than reverting the old custom 'isHex' function which did not validate the existence of prefix '0x', I did the following:
```if (isHexString(addHexPrefix(address))) {
      const checksummedAddress = checksumAddress(addHexPrefix(address));``` which will eventually be encapsulated in the helper method mentioned abovee

Reply:
Makes sense! This looks good